store/store.go

```diff
@@ -899,7 +899,36 @@ func (s *Store) newRequest(reqOptions *requestOptions, user *auth.UserState) (*h
 	req.Header.Set("X-Ubuntu-Series", s.series)
 	req.Header.Set("X-Ubuntu-Classic", strconv.FormatBool(release.OnClassic))
 	req.Header.Set("X-Ubuntu-Wire-Protocol", UbuntuCoreWireProtocol)
+	// still send this for now
 	req.Header.Set("X-Ubuntu-No-CDN", strconv.FormatBool(s.noCDN))
+	if s.noCDN {
+		req.Header.Set("Snap-CDN", "none")
+	} else {
+		if s.authContext != nil {
+			// set Snap-CDN from cloud instance information
+			// if available
+			// TODO: do we want a more complex retry strategy
+			// where we first to send this header and if the
+			// operation fails that way to even get the connection
+			// then we retry without sending this?
+			cloudInfo, err := s.authContext.CloudInfo()
```

Comment:
I wonder if this block should become a small helper for easier reading of `newRequest` - something like: `fund cdnParams(s.authContext()) (string, error)` or similar.

Reply:
can do, I can think later in the cleanups for the new api, this bits should really be done only for Download

Reply:
done